docs/source/04_user_guide/06_pipelines.md

```diff
@@ -533,6 +533,52 @@ kedro run --runner=ParallelRunner
 
 > *Note:* You cannot use both `--parallel` and `--runner` flags at the same time (e.g. `kedro run --parallel --runner=SequentialRunner` raises an exception).
 
+#### Using a custom runner
+
+If the built-in runners don't meet your requirements, you can define your own runner in your project instead. For example, you may want to add a dry runner, a runner which lists which nodes would be run instead of executing them. You can define it in the following way (courtesy of [AntonyMilneQB](https://github.com/AntonyMilneQB])):
+
+```python
+# in your_project/src/your_project/runner.py
+from kedro.io import AbstractDataSet, DataCatalog, MemoryDataSet
+from kedro.pipeline import Pipeline
+from kedro.runner.runner import AbstractRunner
+
+
+class DryRunner(AbstractRunner):
+    """``DryRunner`` is an ``AbstractRunner`` implementation. It can be used to list which
+    nodes would be run without actually executing anything.
+    """
+
+    def create_default_data_set(self, ds_name: str) -> AbstractDataSet:
+        """Factory method for creating the default data set for the runner.
+        Args:
```

Comment:
```suggestion

        Args:
```

Reply:
I would actually consider removing the docstring for this method entirely, seeing as the really interesting part of the code is the `_run` method and this just distracts from it. Unless you want to keep it there for passing linting :)